### PR TITLE
fix: truncate text on mobile (#805) & move dropdown (#806)

### LINKED ISSUE #805: Bug: Mobile view - context filter button has max-w-max, breaks out of container on long repo names and removes the cancel button
### Describe the bug

Long repo names cause the context filter button to break out of its container and lead to unintended horizontal scroll. Also, the cancel button is inaccessible when the component reaches the right margin. 
IMG_3533

### Steps to reproduce

1. Go to https://insights.opensauced.pizza/hot/repositories/filter/JoshuaKGoldberg/expect-no-axe-violations on mobile
2. Look at the header filter component and swipe sideways

### Affected services

insights.opensauced.pizza

### Platforms

Mobile

### Browsers

Chrome

### Environment

Production

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #806: Bug: Mobile view - context filter button filter menu opens off-screen
### Describe the bug

When trying to access the filter menu on mobile, the menu opens off to the side - hiding most of the options

IMG_C453C4A97F38-1

### Steps to reproduce

1. Go to https://insights.opensauced.pizza/hot/repositories/filter/FeroxFoxxo/MQReawaken
2. Click on context filter button in the header
3. filter menu should be partially visible

### Affected services

insights.opensauced.pizza

### Platforms

Mobile

### Browsers

Chrome

### Environment

Production

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by Themesses**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
PR fixes two issues on mobile resolution:

#805 
Filter button breaks out of the container on long repo names. To fix this, the text was truncated with an ellipsis. A further enhancement is possible by removing 'filtered by' or shortening it simply to 'by', if necessary. 

#806 
The drop-down was moved to the left on mobile, to prevent the content from being cut off. 

## Related Tickets & Documents
Fixes #805 
Fixes #806

## Mobile & Desktop Screenshots/Recordings
Before:
https://user-images.githubusercontent.com/43274289/218179334-06e39998-4c53-4176-a917-90f7875807a3.mov

After:

https://user-images.githubusercontent.com/43274289/218180415-6457eb0b-0d69-4135-a338-19aab371e86b.mov

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by Themesses** [comment]
> Looks great @Themesses 👍
> 
> There are some formatting issues that fail the build. Try running `npm run format` and then `npm run build` to check if it's good (no error)
> 

@TAKANOME-DEV if you want to take a look, the build seems to be up. And thanks for the tip!

File: components/molecules/SuperlativeSelector/superlative-selector.tsx [hunk 1]
```diff
@@ -50,33 +49,34 @@ const SuperativeSelector: React.FC<SuperlativeSelectorProps> = ({
...
   return (
-    <div className="max-w-max relative" ref={ref}>
-      <ContextFilterButton onClick={toggleFilter} isSelected={!!selected}>
-        {selected ? (
-          <div className="flex">
-            <div className="flex" onClick={toggleFilter}>
-              <span className="text-dark-slate-10">Filtered by:</span>
-              <div className="flex items-center ml-1 text-light-slate-12">
-                {filterDescription}
+    <div className="!overflow-hidden !text-ellipsis !whitespace-nowrap  md:relative md:!whitespace-normal md:!overflow-visible md:!text-base max-w-max" ref={ref}>
```
Comment (0-vortex) [inline components/molecules/SuperlativeSelector/superlative-selector.tsx:52]: please remove the '!' (important) classes, we cannot manage them :<

**Review by 0-vortex — CHANGES_REQUESTED** [review]
LGTM except the `!important` classes, removing those should do the trick! 👍

**Comment by Themesses** [comment]
> LGTM except the `!important` classes, removing those should do the trick! 👍

@0-vortex love the feedback and thanks! Classes removed.

**Review by bdougie — APPROVED** [review]
🚀

**Comment by 0-vortex** [comment]
> > LGTM except the `!important` classes, removing those should do the trick! 👍
> 
> @0-vortex love the feedback and thanks! Classes removed.

Awesome, can you also remove the commented div? If @bdougie 👍 we can merge it over the weekend too! 🍕

**Comment by Themesses** [comment]
@0-vortex got rid of the commented divs as well as some empty ones that weren't really doing anything.

**Comment by TAKANOME-DEV** [comment]
The deploy preview looks fine on Firefox but not on Brave and Chrome for me :thinking: 
Does anyone noticed the same or it's just mine? 
- Firefox
firefox
- Brave
brave
- Chrome
chrome
cc @Themesses

**Comment by Themesses** [comment]
> The deploy preview looks fine on Firefox but not on Brave and Chrome for me 🤔 Does anyone noticed the same or it's just mine?

Great catch @TAKANOME-DEV .On a Mac, Brave and Firefox seem to work as intended for me. I wonder if it's just simply the medium resolution. I made an assumption that you wouldn't need to truncate on anything other than mobile. Can you give me the screen width where that happens

**Comment by Themesses** [comment]
@TAKANOME-DEV  Here's the new behavior, let me know if that works: 


https://user-images.githubusercontent.com/43274289/218462444-54129e67-7492-46f7-accf-85b7ddd82fd8.mov

File: components/molecules/SuperlativeSelector/superlative-selector.tsx [hunk 2]
```diff
@@ -36,9 +36,9 @@ const SuperativeSelector: React.FC<SuperlativeSelectorProps> = ({
       if (isOpen && ref.current && !ref.current.contains(e.target)) {
         setIsOpen(false);
       }
-
     };
     document.addEventListener("mousedown", checkIfClickedOutside);
+    console.log("current ref", ref.current?.getBoundingClientRect());
```
Comment (brandonroberts) [inline components/molecules/SuperlativeSelector/superlative-selector.tsx:41]: nit: remove console.log

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks @Themesses!

**Pull request merged**